hydromt/drivers/dataframe/pandas_driver.py

```diff
@@ -50,7 +50,7 @@ def read_data(
         extension: str = uri.split(".")[-1]
         logger.info(f"Reading using {self.name} driver from {uri}")
         if extension == "csv":
-            return pd.read_csv(uri, usecols=variables, **self.options)
```

Comment:
Why not usecols?

Reply:
The test uses tests/data/parameters_data.yml data catalog that uses the index_col:0 option. If usecols is used the resulting dataframe does not include the index column. The usecols argument could be added in the data catalog as an option

Reply:
I think the driver should do something with the `variables` keyword. If you request data with only certain variables, we should use the functionality is pandas to filter those out while reading the data.

Reply:
Feel free to improve the `PandasDriver` here!
I also have this workaround in the test here: https://github.com/Deltares/hydromt/blob/v1/tests/drivers/test_pandas_driver.py#L114
If you can make it so that we nicely play with panda's automatically generated indexes / columns you should be able to remove that `if` statement.